pallets/nfts/src/lib.rs

```diff
@@ -1295,80 +1340,128 @@ pub mod pallet {
-			item: T::ItemId,
+			maybe_item: Option<T::ItemId>,
 			delegate: AccountIdLookupOf<T>,
 			maybe_deadline: Option<BlockNumberFor<T>>,
 		) -> DispatchResult {
-			let maybe_check_origin = T::ForceOrigin::try_origin(origin)
-				.map(|_| None)
-				.or_else(|origin| ensure_signed(origin).map(Some).map_err(DispatchError::from))?;
 			let delegate = T::Lookup::lookup(delegate)?;
-			Self::do_approve_transfer(
-				maybe_check_origin,
-				collection,
-				item,
-				delegate,
-				maybe_deadline,
-			)
+			match maybe_item {
+				Some(item) => {
+					let maybe_check_origin =
+						T::ForceOrigin::try_origin(origin).map(|_| None).or_else(|origin| {
+							ensure_signed(origin).map(Some).map_err(DispatchError::from)
+						})?;
+					Self::do_approve_transfer(
+						maybe_check_origin,
+						collection,
+						item,
+						delegate,
+						maybe_deadline,
+					)
+				},
+				None => {
+					let origin = ensure_signed(origin)?;
+					Self::do_approve_collection_transfer(
+						origin,
+						collection,
+						delegate,
+						maybe_deadline,
+					)
+				},
+			}
 		}
 
 		/// Cancel one of the transfer approvals for a specific item.
 		///
 		/// Origin must be either:
 		/// - the `Force` origin;
-		/// - `Signed` with the signer being the Owner of the `item`;
+		/// - `Signed` with the signer being the Owner of the `item` if the `item` is provided;
 		///
 		/// Arguments:
 		/// - `collection`: The collection of the item of whose approval will be cancelled.
-		/// - `item`: The item of the collection of whose approval will be cancelled.
+		/// - `maybe_item`: The optional item of the collection of whose approval will be cancelled.
+		///   If not provided, approval to transfer items in the collection that owned by `origin`
+		///   will be cancelled.
 		/// - `delegate`: The account that is going to loose their approval.
 		///
 		/// Emits `ApprovalCancelled` on success.
 		///
 		/// Weight: `O(1)`
 		#[pallet::call_index(16)]
-		#[pallet::weight(T::WeightInfo::cancel_approval())]
+		#[pallet::weight(T::WeightInfo::cancel_approval(maybe_item.is_some() as u32))]
 		pub fn cancel_approval(
 			origin: OriginFor<T>,
 			collection: T::CollectionId,
-			item: T::ItemId,
+			maybe_item: Option<T::ItemId>,
 			delegate: AccountIdLookupOf<T>,
 		) -> DispatchResult {
-			let maybe_check_origin = T::ForceOrigin::try_origin(origin)
-				.map(|_| None)
-				.or_else(|origin| ensure_signed(origin).map(Some).map_err(DispatchError::from))?;
 			let delegate = T::Lookup::lookup(delegate)?;
-			Self::do_cancel_approval(maybe_check_origin, collection, item, delegate)
+			match maybe_item {
+				Some(item) => {
+					let maybe_check_origin =
+						T::ForceOrigin::try_origin(origin).map(|_| None).or_else(|origin| {
+							ensure_signed(origin).map(Some).map_err(DispatchError::from)
+						})?;
+					Self::do_cancel_approval(maybe_check_origin, collection, item, delegate)
+				},
+				None => {
+					let origin = ensure_signed(origin)?;
```

Comment:
As above, why no force origin consideration?

Reply:
The reason is that permission checks are not enforced in the methods for collection approval. Any signed origin can execute these methods, making the force_origin parameter redundant and causing the origin parameter to be optional without necessity.

Reply:
I think this needs to be changed as well based on my comment also here: https://github.com/r0gue-io/pop-node/pull/387#discussion_r1863361224

ForceOrigin should be able to destroy approvals otherwise a collection owner could maybe be prevented to destroy a collection due to some pending approvals that never get removed. The ForceOrigin, governance, could then destroy those approvals. This does however complicate the removal because we don't have the account id of the owner. This would mean we would have to remove all approvals with the provided collection id, which perhaps has to be protected with a limit (in general there needs to be some limit configured).

Curious what your opinion here is @evilrobot-01

Reply:
> Any signed origin can execute these methods, making the force_origin parameter redundant and causing the origin parameter to be optional without necessity.

This not true based on your comment here: https://github.com/r0gue-io/pop-node/pull/387#discussion_r1863429148